web/packages/teleport/src/Discover/Shared/usePoll.ts

```diff
@@ -7,9 +7,21 @@ export function usePoll<T>(
 ): T | null {
   const abortController = useRef(new AbortController());
   const [result, setResult] = useState<T | null>(null);
+  const [running, setRunning] = useState(false);
 
   useEffect(() => {
-    if (enabled) {
+    if (enabled && !running) {
+      setResult(null);
+      setRunning(true);
+    }
+
+    if (!enabled && running) {
+      setRunning(false);
+    }
```

Comment:
i copy-pasta from your old code, but looking at this block again, it doesn't seem like i need it 🤔